### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Kel-nana**
### set up associations

- Create a class Classroom with the following:
 
  - [ ] @label instance variable, should be initialized in the constructor.
  - [ ]  Setter and getter for @label (remember about attr_accessor).

- Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

  - [ ] Create the has-many side (a classroom has many students).
  - [ ] Create the belongs-to side (a student belongs to a classroom).
  - [ ] Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - [ ] Make sure that when setting the classroom for a student it also adds it to the classrooms' students.

-  Create a class Book with the following:
   - [ ]  @title and @author instance variables, should be initialized in the constructor.
   - [ ] Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:
  - [ ] @date instance variable, should be initialized in the constructor.
  - [ ] Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
    - [ ] Create the has-many side of Book and Rental (a book has many rentals).
    - [ ] Create the belongs-to side of Rental and Book (a rental belongs to a book).
    - [ ] Create the has-many side of Person and Rental (a person has many rentals).
    - [ ] Create the belongs-to side of Rental and Person (a rental belongs to a person).
    - [ ] Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classes/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  # Initializes a new instance of the Book class with the given title and author.
+  # Params:
+  # +title+:: A string representing the title of the book.
+  # +author+:: A string representing the author of the book.
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = [] # Initialize an empty array to store the rentals associated with this book.
+  end
+end
```
Comment (ghost) [inline classes/book.rb]: - [ ] Kindly add a method of adding a rental in book class to establish the `has many` association

```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end

```

**Review by frank1738 — CHANGES_REQUESTED** [review]
Hi @Kel-nana , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @title = title
+    @author = author
+    @rentals = [] # Initialize an empty array to store the rentals associated with this book.
+  end
+
+  # Adds a rental to the list of rentals associated with this book.
+  def add_rental(rental)
+    # @rentals.push(rental)
+    @rentals << rental
+  end
```
Comment (muneebulrehman) [inline classes/book.rb]: The method for adding the rentals should take `book` itself as a parameter along with the date and the person it belongs to. Kindly refer to the example provided by the previous reviewer and update your code accordingly.

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @Kel-nana , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- Good follow-up with the previous reviewer.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/book.rb (hunk 2 above)
Comment (Kel-nana) [reply classes/book.rb]: okay and thank you

File: classes/person.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
+require_relative '../decorators/nameable'
+class Person < Nameable
```
Comment (julie-ify) [inline classes/person.rb:2]: - [x] **[OPTIONAL]** Your Person class does not have the other relationship using `add_rental` as you have in the Book class. I think you should add the same here.

**Review by julie-ify — APPROVED** [review]
Hi @Kel-nana,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/person.rb (hunk 3 above)
Comment (Kel-nana) [reply classes/person.rb:2]: Thank you so much and I have implemented the changes

**Pull request merged**